src/icmp/pinger.cc

```diff
@@ -191,9 +191,7 @@ main(int, char **)
 #endif
 
     for (;;) {
-        struct timeval tv;
-        tv.tv_sec = std::chrono::seconds(PingerTimeout).count();
-        tv.tv_usec = 0;
+        auto tv = ToTimeval(PingerTimeout);
```

Comment:
The new converter handles `tv_usec` which adds a performance regression calculating it to be `0` on every loop iteration. That being the only real "gain" pushes this into the "not valueable yet" type of changes for me.

Reply:
> The new converter handles `tv_usec` which adds a performance regression calculating it to be `0` on every loop iteration. 

AFAICT, the above statement is based on two false assumptions:

False assumption A: Proposed compiled code will be different from official compiled code. In reality, modern optimizing compilers see enough in this context to generate [identical assembly](https://godbolt.org/z/vKdf5Woq6) in both cases (even at [-O1](https://godbolt.org/z/c6b5n17To) optimization level)!

False assertion B: This loop is performance sensitive (at the precision level of a few integer operations added or removed). In reality, this is an I/O-waiting loop in a pinger helper. Adding a few quick ops to that loop would not change pinger performance, even under heavy load.

In summary, this PR does not "add a performance regression".

N.B. Proposed loop code is also clearly better -- it is a bit easier to read, and it supports reasonable PingerTimeout values that the official code silently mishandles. We may never need to support fractional seconds in this context, but better code is still an improvement.

> That being the only real "gain" pushes this into the "not valueable yet" type of changes for me.

Please do not block PRs just because you do not see value in them (when other core developers do see value).